### ISSUE TITLE
AMZN Linux Docker version mismatch: 18.06.1ce-7.25.amzn1 and 18.06.1ce-8.28.amzn1.x86_64

### ISSUE THREAD

**Issue by amyweiyou** [NVIDIA/nvidia-docker#945]
_The template below is mostly useful for bug reports and support questions. Feel free to remove anything which doesn't apply to you and add more information where it makes sense._

_Also, before reporting a new issue, please make sure that:_

- _You read carefully the [documentation and frequently asked questions](https://github.com/NVIDIA/nvidia-docker/wiki)._
- _You [searched](https://github.com/NVIDIA/nvidia-docker/issues?utf8=%E2%9C%93&q=is%3Aissue) for a similar issue and this is not a duplicate of an existing one._
- _This issue is not related to [NGC](https://github.com/NVIDIA/nvidia-docker/wiki/NGC), otherwise, please use the [devtalk forums](https://devtalk.nvidia.com/default/board/200/nvidia-gpu-cloud-ngc-users/) instead._
- _You went through the [troubleshooting](https://github.com/NVIDIA/nvidia-docker/wiki/Troubleshooting) steps._

---

### 1. Issue or feature description
Docker version in nvidia-docker2 and the one in Amazon ECS Optimized AMI are a mismatch when trying to build nvidia-docker2 into the AMI
### 3. Information to [attach](https://help.github.com/articles/file-attachments-on-issues-and-pull-requests/) (optional if deemed irrelevant)
When building nvidia-docker2 into the ECS optimized AMI, I got errors below: 
[docker]     us-west-2: ---> Package nvidia-docker2.noarch 0:2.0.3-7.docker18.06.1ce.amzn1 will be installed
[docker]     us-west-2: --> Processing Dependency: docker = 18.06.1ce-7.25.amzn1 for package: nvidia-docker2-2.0.3-7.docker18.06.1ce.amzn1.noarch
[docker]     us-west-2: --> Finished Dependency Resolution
                                                          [docker] 
[docker]     us-west-2: Error: Package: nvidia-docker2-2.0.3-7.docker18.06.1ce.amzn1.noarch (nvidia-docker)
                           [docker] 
[docker]     us-west-2:            Requires: docker = 18.06.1ce-7.25.amzn1
                                                                          [docker] 
[docker]     us-west-2:            Installed: docker-18.06.1ce-8.28.amzn1.x86_64 (installed)
[docker]     us-west-2:                docker = 18.06.1ce-8.28.amzn1
                                                                    [docker] 
[docker]     us-west-2:            Available: docker-17.12.1ce-1.135.amzn1.x86_64 (amzn-main)
             [docker] 
[docker]     us-west-2:                docker = 17.12.1ce-1.135.amzn1
                                                                     [docker] 
[docker]     us-west-2:            Available: docker-18.03.1ce-2.141.amzn1.x86_64 (amzn-updates)
                [docker] 
[docker]     us-west-2:                docker = 18.03.1ce-2.141.amzn1
                                                                     [docker] 
[docker]     us-west-2:            Available: docker-18.03.1ce-5.0.amzn1.x86_64 (amzn-updates)
              [docker] 
[docker]     us-west-2:                docker = 18.03.1ce-5.0.amzn1
                                                                   [docker] 
[docker]     us-west-2:            Available: docker-18.06.1ce-2.16.amzn1.x86_64 (amzn-updates)
               [docker] 
[docker]     us-west-2:                docker = 18.06.1ce-2.16.amzn1
[docker]     us-west-2:            Available: docker-18.06.1ce-3.17.amzn1.x86_64 (amzn-updates)
[docker]     us-west-2:                docker = 18.06.1ce-3.17.amzn1
[docker]     us-west-2:            Available: docker-18.06.1ce-4.19.amzn1.x86_64 (amzn-updates)
[docker]     us-west-2:                docker = 18.06.1ce-4.19.amzn1
[docker]     us-west-2:            Available: docker-18.06.1ce-5.22.amzn1.x86_64 (amzn-updates)
               [docker] 
    us-west-2:                docker = 18.06.1ce-5.22.amzn1
                                                           [docker] 
[docker]     us-west-2:            Available: docker-18.06.1ce-6.24.amzn1.x86_64 (amzn-updates)
[docker]     us-west-2:                docker = 18.06.1ce-6.24.amzn1
[docker]     us-west-2:            Available: docker-18.06.1ce-7.25.amzn1.x86_64 (amzn-updates)
    us-west-2:                docker = 18.06.1ce-7.25.amzn1

**Comment by nskool** [comment]
On an Amazon Linux AMI with the following packages installed:

```
$ sudo yum list installed | grep docker
docker.x86_64                        18.06.1ce-7.25.amzn1          @amzn-updates
nvidia-container-runtime.x86_64      2.0.0-7.docker18.06.1ce.amzn1 @nvidia-container-runtime
nvidia-docker2.noarch                2.0.3-7.docker18.06.1ce.amzn1 @nvidia-docker
```
Running `sudo yum update` breaks as:
```
--> Running transaction check
---> Package docker.x86_64 0:18.06.1ce-7.25.amzn1 will be updated
--> Processing Dependency: docker = 18.06.1ce-7.25.amzn1 for package: nvidia-docker2-2.0.3-7.docker18.06.1ce.amzn1.noarch
---> Package docker.x86_64 0:18.06.1ce-8.28.amzn1 will be an update
.
.
.
---> Package libnvidia-container-tools.x86_64 0:1.0.0-1 will be updated
---> Package libnvidia-container-tools.x86_64 0:1.0.2-1 will be an update
---> Package libnvidia-container1.x86_64 0:1.0.0-1 will be updated
---> Package libnvidia-container1.x86_64 0:1.0.2-1 will be an update
.
.
.
--> Finished Dependency Resolution
Error: Package: nvidia-docker2-2.0.3-7.docker18.06.1ce.amzn1.noarch (@nvidia-docker)
           Requires: docker = 18.06.1ce-7.25.amzn1
           Removing: docker-18.06.1ce-7.25.amzn1.x86_64 (@amzn-updates)
               docker = 18.06.1ce-7.25.amzn1
           Updated By: docker-18.06.1ce-8.28.amzn1.x86_64 (amzn-updates)
               docker = 18.06.1ce-8.28.amzn1
           Available: docker-17.12.1ce-1.135.amzn1.x86_64 (amzn-main)
               docker = 17.12.1ce-1.135.amzn1
           Available: docker-18.03.1ce-2.141.amzn1.x86_64 (amzn-updates)
               docker = 18.03.1ce-2.141.amzn1
           Available: docker-18.03.1ce-5.0.amzn1.x86_64 (amzn-updates)
               docker = 18.03.1ce-5.0.amzn1
           Available: docker-18.06.1ce-2.16.amzn1.x86_64 (amzn-updates)
               docker = 18.06.1ce-2.16.amzn1
           Available: docker-18.06.1ce-3.17.amzn1.x86_64 (amzn-updates)
               docker = 18.06.1ce-3.17.amzn1
           Available: docker-18.06.1ce-4.19.amzn1.x86_64 (amzn-updates)
               docker = 18.06.1ce-4.19.amzn1
           Available: docker-18.06.1ce-5.22.amzn1.x86_64 (amzn-updates)
               docker = 18.06.1ce-5.22.amzn1
           Available: docker-18.06.1ce-6.24.amzn1.x86_64 (amzn-updates)
               docker = 18.06.1ce-6.24.amzn1
 You could try using --skip-broken to work around the problem
 You could try running: rpm -Va --nofiles --nodigest
[ec2-user@ip-172-31-79-119 ~]$ 
```

**Comment by RenaudWasTaken** [comment]
Sorry for the delay, I took some time to dig into the amazon linux packaging scheme this time around. Turns out my understanding wasn't correct.

I published new packages which should ensure that any subsequent **Releases** of 18.06.1 on amazon linux **will not** break.

This will however continue to happen for any subsequent **Versions** of docker as there is still a hard dependency on the docker version (namely the runc version associated with it) which we won't remove as this prevents undefined behavior.

**Comment by amyweiyou** [comment]
Hi Renaud,
Thank you so much for the quick action! The build succeeded now!
It seems that the new package takes the nvidia-container-runtime-2.0.0-1.docker18.06.1.amzn1.x86_64.rpm as dependency, instead of the latest nvidia-container-runtime-2.0.0-7.docker18.06.1ce.amzn1.x86_64.rpm. Is there a reason?
